src/odemis/driver/actuator.py

```diff
@@ -1071,21 +1055,33 @@ def moveAbs(self, pos):
 
         return f
 
-    def _doMoveAbs(self, pos):
-        axis, distance = pos.items()[0]
-
+    def _referenceIfNeeded(self):
+        """
+        Force a referencing if it has moved a certain number of times
+        """
         # Reference from time to time
         self._move_num += 1
         if self._ref_period and self._move_num >= self._ref_period:
```

Comment:
Should we update the counter after the if block? In this case we actually reference **after** n moves and not **on**  the nth move as specified.

Also, is it still going to work if _ref_period is 0? moveabssync prevents infinite recursions, so I guess it's fine, but we should probably at least add a test case for that.